cmd/contact-auditor/main.go

```diff
@@ -7,54 +7,52 @@ import (
 	"flag"
 	"fmt"
 	"io/ioutil"
+	"os"
 	"strings"
 	"time"
 
-	"github.com/jmhodges/clock"
+	"github.com/go-sql-driver/mysql"
 	"github.com/letsencrypt/boulder/cmd"
 	blog "github.com/letsencrypt/boulder/log"
 	"github.com/letsencrypt/boulder/policy"
+	"github.com/letsencrypt/boulder/test/vars"
 )
 
-var queryInterrupted bool
-
 type contactAuditor struct {
-	db     *sql.DB
-	logger blog.Logger
-	clk    clock.Clock
-	grace  time.Duration
+	db            *sql.DB
+	resultsFile   *os.File
+	writeToStdout bool
+	logger        blog.Logger
 }
 
 type result struct {
-	// Receiver for the `id` column.
-	ID int64
-
-	// Receiver for the `contact` column.
-	Contact  []byte
-	contacts []string
+	id        int64
+	contacts  []string
+	createdAt string
 }
 
-func (r *result) unmarshalContact() error {
-	var contact []string
-	err := json.Unmarshal(r.Contact, &contact)
+func unmarshalContact(contact []byte) ([]string, error) {
+	var contacts []string
+	err := json.Unmarshal(contact, &contacts)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	r.contacts = append(r.contacts, contact...)
-	return nil
+	return contacts, nil
 }
 
-func (r *result) validateContact() error {
+func validateContacts(id int64, createdAt string, contacts []string) error {
 	// Setup a buffer to store any validation problems we encounter.
 	var probsBuff strings.Builder
 
 	// Helper to write validation problems to our buffer.
 	writeProb := func(contact string, prob string) {
 		// Add validation problem to buffer.
-		probsBuff.WriteString(fmt.Sprintf("[ %q: %q ] ", contact, prob))
+		probsBuff.WriteString(
+			fmt.Sprintf("%d\t%s\tvalidation\t%q\t%q\n", id, createdAt, contact, prob),
+		)
```

Comment:
I don't consider this a blocker, but thought you might like to know: since `strings.Builder` is a `io.Writer` (it has a `Write` method with the right signature), you can use `fmt.Fprintf`: https://golang.org/pkg/fmt/#Fprintf

    fmt.Fprintf(probsBuff, "%d\t%s\tvalidation\t%q\t%q\n", id, createdAt, contact, prob)

Reply:
Oooh this is very cool. Thanks!